changelog.d/11217.bugfix

```diff
@@ -0,0 +1 @@
+Fixes a bug introduced in #9958 which made it impossible to join rooms that return a send_join response containing floats.
```

Comment:
```suggestion
Fix a bug introduced in 1.35.0 which made it impossible to join rooms that return a `send_join` response containing floats.
```

The release containing a PR can be found by clicking on the squash commit and looking through the list of tags
https://github.com/matrix-org/synapse/commit/64887f06fcac63e069364d625d984b4951bf1ffc